app/models/course/assessment/question/text_response.rb

```diff
@@ -14,6 +14,8 @@ class Course::Assessment::Question::TextResponse < ApplicationRecord
 
   accepts_nested_attributes_for :groups, allow_destroy: true
 
+  after_initialize :build_at_least_one_group, if: -> { :comprehension_question? && :new_record? }
```

Comment:
`after_initialize` will have unintended effects.

For a simple example, try creating a question using the defaults (with one group, one point, no solutions; i.e. 1) go to new question page, 2) fill in title and max grade, 3) submit), two groups are created even though one was submitted.

If the intent is to initialize it for a new question form, perhaps shift this into a method by itself and manually call it in the new action?